internal/kafka/internal/services/quota/ams_quota_service_test.go

```diff
@@ -48,6 +48,407 @@ var ocmClientMockWithCloudAccounts = &ocm.ClientMock{
+						if product != string(ocm.RHOSAKProduct) {
+							return []*v1.QuotaCost{}, nil
+						}
+						rrbq1 := v1.NewRelatedResource().
+							BillingModel(string(v1.BillingModelStandard)).
+							Product(string(ocm.RHOSAKProduct)).
+							ResourceName(resourceName).
+							Cost(1)
+
+						rrbq2 := v1.NewRelatedResource().
+							BillingModel(string(v1.BillingModelMarketplace)).
+							Product(string(ocm.RHOSAKProduct)).
+							ResourceName(resourceName).
+							Cost(1)
+
+						qcb, err := v1.NewQuotaCost().Allowed(1).Consumed(0).OrganizationID(organizationID).RelatedResources(rrbq1, rrbq2).CloudAccounts(
+							v1.NewCloudAccount().CloudProviderID("xyz").CloudAccountID("1234567890"),
+						).Build()
+						if err != nil {
+							panic("unexpected error")
+						}
+
+						return []*v1.QuotaCost{qcb}, nil
+					},
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr:          false,
+			wantBillingModel: string(v1.BillingModelStandard),
+		},
+		{
+			name: "marketplace billing is selected when billing account is provided",
+			args: args{
+				request: dbapi.KafkaRequest{
+					Name:                  "test",
+					SizeId:                "x1",
+					InstanceType:          "standard",
+					BillingCloudAccountId: "1234567890",
+					Marketplace:           "aws",
+				},
+			},
+			fields: fields{
+				ocmClient: &ocm.ClientMock{
+					ClusterAuthorizationFunc: func(cb *v1.ClusterAuthorizationRequest) (*v1.ClusterAuthorizationResponse, error) {
+						ca, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Build()
+						return ca, nil
+					},
+					GetOrganisationIdFromExternalIdFunc: func(externalId string) (string, error) {
+						return fmt.Sprintf("fake-org-id-%s", externalId), nil
+					},
+					GetQuotaCostsForProductFunc: func(organizationID, resourceName, product string) ([]*v1.QuotaCost, error) {
+						if product != string(ocm.RHOSAKProduct) {
+							return []*v1.QuotaCost{}, nil
+						}
+						rrbq1 := v1.NewRelatedResource().
+							BillingModel(string(v1.BillingModelStandard)).
+							Product(string(ocm.RHOSAKProduct)).
+							ResourceName(resourceName).
+							Cost(1)
+
+						rrbq2 := v1.NewRelatedResource().
+							BillingModel(string(v1.BillingModelMarketplace)).
+							Product(string(ocm.RHOSAKProduct)).
+							ResourceName(resourceName).
+							Cost(1)
+
+						qcb, err := v1.NewQuotaCost().Allowed(1).Consumed(0).OrganizationID(organizationID).RelatedResources(rrbq1, rrbq2).CloudAccounts(
+							v1.NewCloudAccount().CloudProviderID("aws").CloudAccountID("1234567890"),
+						).Build()
+						if err != nil {
+							panic("unexpected error")
+						}
+
+						return []*v1.QuotaCost{qcb}, nil
+					},
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr:          false,
+			wantBillingModel: string(v1.BillingModelMarketplaceAWS),
+		},
+	}
+
+	RegisterTestingT(t)
+	for _, tt := range tests {
```

Comment:
I think the same pattern as here: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1089/files#diff-ace4f53881679319ea877b48a99b7466398f3187ef68ae0825bb1ba233632168R49 needs to be followed for the `tt` in the tests

Reply:
good catch, fixed